cellar_rebalancer/src/commands/cosmos_mode.rs

```diff
@@ -2,111 +2,56 @@
-
-        let eth_host = config.ethereum.rpc.clone();
-
-        let mut pollers = vec![];
-
-        for cellar in config.cellars.clone().into_iter() {
-            let client = Provider::<Http>::try_from(eth_host.clone()).unwrap();
-            let client = SignerMiddleware::new(client, wallet.clone());
-            let client = Arc::new(client);
-            let mongo = config.mongo.clone();
-
-            let name = &config.keys.rebalancer_key;
-            let cosmos_key = config.load_deep_space_key(name.clone());
-
-            let poller = Poller::new(&cellar, client, &mongo, &cosmos_key, config.clone())
-                .await
-                .unwrap_or_else(|e| {
-                    status_err!("couldn't initialize poller: {}", e);
-                    std::process::exit(1);
-                });
-
-            pollers.push(poller);
-        }
-
-        pollers
-    }
-}
+pub struct CosmosSignerCmd;
 
 impl Runnable for CosmosSignerCmd {
-    /// Start the cosmos mode signer.
+    /// Start the application.
     fn run(&self) {
-        info!("Starting application");
-
         let config = APP.config();
+        info!("Starting application");
         abscissa_tokio::run(&APP, async {
-            let collector =
-                ServiceBuilder::new()
-                    .buffer(20)
-                    .service(Collector::new(&config).unwrap_or_else(|e| {
-                        status_err!("couldn't initialize collector service: {}", e);
-                        std::process::exit(1);
-                    }));
-
-            let pollers = self.build_pollers(config.as_ref().clone()).await;
-
-            let mut tasks = vec![];
-            for poller in pollers {
-                let collector = collector.clone();
-                let task = tokio::spawn(async move { poller.run(collector).await });
-                tasks.push(task);
-            }
-            future::join_all(tasks).await;
+            // Reflection required for certain clients to function... such as grpcurl
+            let contents = fs::read("steward_proto/src/prost/descriptor.bin").unwrap();
+            let proto_descriptor_service = tonic_reflection::server::Builder::configure()
+                .register_encoded_file_descriptor_set(contents.as_slice())
+                .build()
+                .unwrap();
+
+            // Configure TLS
+            let cert = tokio::fs::read(&config.tls.server_cert).await.unwrap();
+            let key = tokio::fs::read(&config.tls.server_key).await.unwrap();
+            let server_identity = Identity::from_pem(cert, key);
+            let client_ca_cert = tokio::fs::read(&config.tls.client_ca_cert).await.unwrap();
+            let client_ca_cert = Certificate::from_pem(client_ca_cert);
+            let tls_config = ServerTlsConfig::new()
+                .identity(server_identity)
+                .client_ca_root(client_ca_cert);
+
+            // run it
+            let addr = ([127, 0, 0, 1], 3000).into();
```

Comment:
We may want to adjust this port assignment, it's the default port for a lot of webservers in test. We could also make it a config option but we'd have to ensure data providers would be connecting to a full URL including the port.